File: hw/toapanta/u1/hw16IncomeTaxLib/Java/src/ec/edu/espe/incometax/view/IncomeTaxCalculator.java
package ec.edu.espe.incometax.view;

/**
 *
 * @author Adrian Toapanta, Object Masters, @ESPE
 */
import ec.edu.espe.incometaxcalculation.IncomeTax;
import java.util.Scanner;

public class IncomeTaxCalculator {
    public static void main(String[] args) {
         Scanner scanner = new Scanner(System.in);
    
        double monthlySalary = IncomeTax.getValidatedDouble(scanner, "your  salary in DOLLARS");
        double monthlyExpenses = IncomeTax.getValidatedDouble(scanner, "your monthly expenses");
        int institutionalSector = IncomeTax.getValidatedSector(scanner);
        
        IncomeTax.showTaxResults(monthlySalary, institutionalSector, monthlyExpenses);
    
        scanner.close();
    }
}

File: hw/toapanta/u1/hw16IncomeTaxLib/JS/ec/edu/espe/incometax/view/IncomeTaxCalculator.js
const { IncomeTax } = require('../calculation/IncomeTax.js');

class IncomeTaxCalculator {
    static async main() {
        console.log("        INCOME TAX CALCULATOR BY KEVIN CHALAN");
        
        const monthlySalary = await IncomeTax.getValidatedDouble("your salary in DOLLARS: ");
        const monthlyExpenses = await IncomeTax.getValidatedDouble("your monthly expenses: ");
        const institutionalSector = await IncomeTax.getValidatedSector();
        
        IncomeTax.showTaxResults(monthlySalary, institutionalSector, monthlyExpenses);
        
        process.exit(0);
    }
}

// Ejecutar la aplicación
IncomeTaxCalculator.main();